### PR TITLE
Feature/tao 2783 session history

### PR DESCRIPTION
**Pull request by jsconan**
Related to: https://oat-sa.atlassian.net/browse/TAO-2783

### CODE REVIEW

**Comment by jsconan** [comment]
This is a big PR.
@antoinerobin will verify the PHP.
@krampstudio please check the JavaScript part and do the merge when all will be ok.

**Comment by krampstudio** [comment]
@jsconan I'll prefer to only review the code, otherwise I won't have the time to do it before tomorrow morning.

File: views/js/controller/Delivery/monitoring.js [hunk 1]
```diff
@@ -119,40 +121,79 @@ define([
...
+            // display the session history
+            function showHistory(selection) {
+                var urlParams = {
+                    testCenter : testCenterId,
+                    session: selection
+                };
+                if (deliveryId) {
+                    urlParams.delivery = deliveryId;
+                }
+                location.href = helpers._url('sessionHistory', 'Reporting', 'taoProctoring', urlParams);
```
Comment (krampstudio) [inline views/js/controller/Delivery/monitoring.js]: is it the `location` from `window.location` ? If yes, please specify it (so everybody knows where this is from).

File: views/js/component/history/historyTable.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                    icon: 'reset',
+                    title: __('Refresh the page'),
+                    label: __('Refresh'),
+                    action: function() {
+                        self.refresh();
+                    }
+                }];
+                this.getElement()
+                    .on('query.datatable', function() {
+                        loadingBar.start();
```
Comment (krampstudio) [inline views/js/component/history/historyTable.js]: I wouldn't handle the loading bar from within the component but from the controller: trigger events, like `loading` and `loaded` and the controller will handle the loading bar, in order to decouple components from their context.

**Comment by krampstudio** [comment]
except the small remarks, this looks good for me. You can go ahead.

**Pull request merged**